airbyte-integrations/connectors/source-sftp/src/main/java/io/airbyte/integrations/source/sftp/SftpCommand.java

```diff
@@ -0,0 +1,133 @@
+                .map(this::transformFileExtension)
+                .collect(Collectors.toSet());
+        filePattern = Pattern.compile(config.has("file_pattern") ? config.get("file_pattern").asText() : "");
+    }
+
+    private SupportedFileExtension transformFileExtension(String fileExtension) {
+        try {
+            return SupportedFileExtension.valueOf(fileExtension.toUpperCase());
+        } catch (Exception e) {
+            LOGGER.error("Unsupported file extension : {}", fileExtension);
+            throw new RuntimeException();
+        }
+    }
+
+    public void tryChangeWorkingDirectory(String path) {
+        try {
+            checkIfConnected();
+            client.changeWorkingDirectory(FilenameUtils.normalize(path));
+            LOGGER.info("Working directory set to : {}", path);
+        } catch (SftpException e) {
+            LOGGER.error("Could not find path : {} : ", path, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Map<String, JsonNode> getFilesSchemas() {
+        checkIfConnected();
+        Map<String, JsonNode> fileSchemas = new HashMap<>();
+        Set<String> fileNames = getFileNames();
+        LOGGER.info("Found file for sync : {}", fileNames);
+        fileNames.forEach(fileName -> {
+            ByteArrayInputStream file = client.getFile(fileName);
+            JsonNode parsedFile = tryGetFirstNode(file, fileName);
+            JsonNode schema = tryGetSchema(parsedFile, fileName);
+            fileSchemas.put(fileName, schema);
+        });
+        return fileSchemas;
+    }
+
+    private Set<String> getFileNames() {
+        checkIfConnected();
+        Vector<LsEntry> entries = new Vector<>();
+        for (SupportedFileExtension fileExtension : selectedFileExtensions) {
+            entries.addAll(client.lsFile(fileExtension));
+        }
+        return entries.stream()
+                .map(ChannelSftp.LsEntry::getFilename)
+                .filter(filePattern.asPredicate())
+                .collect(Collectors.toSet());
+    }
+
+    private JsonNode tryGetSchema(JsonNode jsonNode, String fileName) {
+        try {
+            return JsonSchemaGenerator.getJsonSchema(jsonNode);
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while trying to parse schema of file {} : ", fileName, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public JsonNode tryGetFirstNode(ByteArrayInputStream file, String fileName) {
+        try {
+            String extension = FilenameUtils.getExtension(fileName);
+            SftpFileParser parser = sftpFileParserFactory.create(transformFileExtension(extension));
+            return parser.parseFileFirstEntity(file);
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while trying to parse file {} : ", fileName, e);
+            throw new RuntimeException();
+        }
+    }
+
+    public List<JsonNode> getFileData(String fileName) {
```

Comment:
Maybe I am missing something, but I don't understand why there is a need for a separate `tryParseFile` private method instead of having all the code of `tryParseFile` inside `getFileData`

Reply:
The idea was to make two steps visible. 1 - get file 2 - parse this file. But after looking on this second time I guess you're right and I'll put it in one `getFileData` method